src/core/data.jl

```diff
@@ -1454,6 +1454,17 @@ function check_status(data::Dict{String,<:Any})
     end
 end
 
+"get the reference bus in a network dataset"
+function reference_bus(data::Dict{String,<:Any})
+    ref_buses = [bus for (i,bus) in data["bus"] if bus["bus_type"] == 3]
+
+    if length(ref_buses) != 1
+        Memento.error(_LOGGER, "exactly one refrence bus in data is required when calling reference_bus, given $(length(ref_buses))")
```

Comment:
not sure what the best habit is, but I would rather have the calling function check if there is only one reference bus?

Reply:
I was thinking another function, say `reference_buses` could be used to get all buses, while `reference_bus` would only be used when you assume there is exactly one bus.  Sound reasonable?

Reply:
I think it is reasonable. It will also allow for the future implementation of a distributed slack.